File: frmSimple.vb
Option Strict Off
Option Explicit On
Friend Class frmSimple
	Inherits System.Windows.Forms.Form
	
	Private Sub cmdStart_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdStart.Click
		Status = "Loading Lot Information for LOT " & txtLotNumber.Text
		cmdStart.Enabled = False
		Call LotINFO.LoadLotInfo((txtLotNumber.Text))
		LotINFO.TestSetup = "c:\a.mds"
		Call LoadTestSetup((LotINFO.TestSetup))
		'BarDriver.AutomationEvents.TotalBars = LotINFO.LotSize
		'BarDriver.AutomationEvents.MoveToNthBar (LotINFO.FirstBar)
		'BarDriver.AutomationEvents.BarType = LotINFO.BarType
		'qst.SystemParameters.lotID = txtLotNumber.Text
		''Open a new log file
		qst.OptionsParameters.LogFileType = LOGCSV 'log file type definitions in globals
		qst.OptionsParameters.LogFileName = "c:\setups\" & LotINFO.TestSetup & CDate(Now) & LotINFO.lotID & " - NEW.csv"
		Call qst.QuasiParameters.OpenLogFile(qst.OptionsParameters.LogFileName)
		
		Status = "Starting the Test for LOT " & txtLotNumber.Text
		qst.QuasiParameters.Start = True
	End Sub
	
	Private Sub cmdStop_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles cmdStop.Click
		cmdStart.Enabled = True
		qst.QuasiParameters.AbortTest = True
	End Sub
	
	Private Sub frmSimple_Load(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles MyBase.Load
		Call EnableControls(False)
	End Sub
	
	Private Sub frmSimple_FormClosing(ByVal eventSender As System.Object, ByVal eventArgs As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
		QuasiEvents = Nothing
		Call qst.CloseApplication()
		qst = Nothing
		Exit Sub
	End Sub
	
	Public WriteOnly Property Status() As String
		Set(ByVal Value As String)
			lstStatus.Items.Insert(0, Value)
		End Set
	End Property
	
	Private Sub txtLotNumber_Click(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtLotNumber.Click
		txtLotNumber.SelectionStart = 0
		txtLotNumber.SelectionLength = Len(txtLotNumber.Text)
	End Sub
	
	Private Sub txtLotNumber_Enter(ByVal eventSender As System.Object, ByVal eventArgs As System.EventArgs) Handles txtLotNumber.Enter
		txtLotNumber.SelectionStart = 0
		txtLotNumber.SelectionLength = Len(txtLotNumber.Text)
	End Sub
	
	Public Sub EnableControls(Optional ByRef DoEnable As Boolean = True)
		txtLotNumber.Enabled = DoEnable
		cmdStart.Enabled = DoEnable
		cmdStop.Enabled = DoEnable
	End Sub
End Class

File: CS/frmSimple.cs
using System;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using System.Collections;

namespace RemoteOp
{
	partial class frmSimple : System.Windows.Forms.Form
	{
		
		public void cmdStart_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			Status = "Loading Lot Information for LOT " + txtLotNumber.Text;
			cmdStart.Enabled = false;
			string temp_LotNumber = (txtLotNumber.Text);
			modMain.LotINFO.LoadLotInfo(ref temp_LotNumber);
			modMain.LotINFO.TestSetup = "c:\\a.mds";
			modMain.LoadTestSetup(modMain.LotINFO.TestSetup);
			//BarDriver.AutomationEvents.TotalBars = LotINFO.LotSize
			//BarDriver.AutomationEvents.MoveToNthBar (LotINFO.FirstBar)
			//BarDriver.AutomationEvents.BarType = LotINFO.BarType
			//qst.SystemParameters.lotID = txtLotNumber.Text
			//'Open a new log file
			modMain.qst.OptionsParameters.LogFileType = modMain.LOGCSV; //log file type definitions in globals
			modMain.qst.OptionsParameters.LogFileName = "c:\\setups\\" + modMain.LotINFO.TestSetup + System.Convert.ToString(DateTime.Now) + modMain.LotINFO.lotID + " - NEW.csv";
            string lfn = modMain.qst.OptionsParameters.LogFileName;
            string ret = modMain.qst.QuasiParameters.OpenLogFile(ref lfn);
			
			Status = "Starting the Test for LOT " + txtLotNumber.Text;
			modMain.qst.QuasiParameters.Start = true;
		}
		
		public void cmdStop_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			cmdStart.Enabled = true;
			modMain.qst.QuasiParameters.AbortTest = true;
		}
		
		public void frmSimple_Load(System.Object eventSender, System.EventArgs eventArgs)
		{
			EnableControls(false);
		}
		
		public void frmSimple_FormClosing(System.Object eventSender, System.Windows.Forms.FormClosingEventArgs eventArgs)
		{
			modMain.QuasiEvents = null;
			modMain.qst.CloseApplication();
			modMain.qst = null;
			return;
		}
		
public string Status
		{
			set
			{
				lstStatus.Items.Insert(0, value);
			}
		}
		
		public void txtLotNumber_Click(System.Object eventSender, System.EventArgs eventArgs)
		{
			txtLotNumber.SelectionStart = 0;
			txtLotNumber.SelectionLength = txtLotNumber.Text.Length;
		}
		
		public void txtLotNumber_Enter(System.Object eventSender, System.EventArgs eventArgs)
		{
			txtLotNumber.SelectionStart = 0;
			txtLotNumber.SelectionLength = txtLotNumber.Text.Length;
		}
		
		public void EnableControls(bool DoEnable)
		{
			txtLotNumber.Enabled = DoEnable;
			cmdStart.Enabled = DoEnable;
			cmdStop.Enabled = DoEnable;
		}
	}
}
